elasticdl/python/master/master.py

```diff
@@ -0,0 +1,401 @@
+
+        return tb_service
+
+    def _create_checkpoint_service(self, args):
+        checkpoint_service = None
+        if (
+            args.checkpoint_steps
+            or self.job_type == JobType.TRAINING_WITH_EVALUATION
+        ):
+            self.logger.info("Create checkpoint service")
+            checkpoint_service = CheckpointService(
+                args.checkpoint_dir,
+                args.checkpoint_steps,
+                args.keep_checkpoint_max,
+                self.job_type == JobType.TRAINING_WITH_EVALUATION,
+            )
+
+        return checkpoint_service
+
+    def _create_evaluation_service(self, args):
+        evaluation_service = None
+        if (
+            self.job_type == JobType.TRAINING_WITH_EVALUATION
+            or self.job_type == JobType.EVALUATION_ONLY
+        ):
+            self.logger.info(
+                "Create evaluation service with throttle seconds %d "
+                " and evaluation steps %d",
+                args.evaluation_throttle_secs,
+                args.evaluation_steps,
+            )
+            evaluation_service = EvaluationService(
+                self.checkpoint_service,
+                self.tb_service,
+                self.task_d,
+                args.evaluation_start_delay_secs,
+                args.evaluation_throttle_secs,
+                args.evaluation_steps,
+                self.job_type == JobType.EVALUATION_ONLY,
+                self.model_module[args.eval_metrics_fn],
+            )
+            self.task_d.set_evaluation_service(evaluation_service)
+
+        return evaluation_service
+
+    def _create_embedding_service(self, args):
+        endpoint = None
+        embedding_dims = {}
+
+        # Search for embedding layers in the model,
+        # if found, initialize embedding service
+        layers = find_layer(self.model_inst, Embedding)
+        if layers:
+            embedding_service = EmbeddingService()
+            endpoint = embedding_service.start_embedding_service(
+                job_name=args.job_name,
+                image_name=args.worker_image,
+                namespace=args.namespace,
+                resource_request=args.master_resource_request,
+                resource_limit=args.master_resource_limit,
+                pod_priority=args.worker_pod_priority,
+                volume=args.volume,
+                image_pull_policy=args.image_pull_policy,
+                restart_policy=args.restart_policy,
+                cluster_spec=args.cluster_spec,
+            )
+            self.logger.info(
+                "Embedding service start succeeded. The endpoint is %s."
+                % str(endpoint)
+            )
+            embedding_dims = dict(
+                [(layer.name, layer.output_dim) for layer in layers]
+            )
+
+        return endpoint, embedding_dims
+
+    def _create_master_service(self, args):
+        # The master service
+        self.logger.info("Create master service")
```

Comment:
Create -> Creating

Reply:
Good suggestion, updated the similar cases.